### ISSUE TITLE
NoSuchModuleError: Can't load plugin: sqlalchemy.dialects:duckdb

### ISSUE THREAD

**Issue by djouallah** [ibis-project/ibis#4558]
first time trying ibis, using windows 10 and ibis 3.2

```
import ibis
con = ibis.duckdb.connect()  # in-memory database
con.list_tables()
```
 I get the error in the title.
I got confused, although duckdb is the default back-end, it has to be installed separately

**Comment by cpcloud** [comment]
@djouallah Thanks for opening an issue.

Yes, duckdb is the default backend for in-memory operations, however it's still an optional dependency (an `extra` in setuptools lingo).

Check out the [DuckDB backend installation docs](https://ibis-project.org/docs/3.2.0/backends/DuckDB/#install) to learn how to do that!

**Comment by cpcloud** [comment]
I'm going to close this out! Please open a new issue if you get the same error after taking the appropriate installation steps.

**Comment by rpmeira** [comment]
Hi, I have the same error here, even though I have duckdb installed.
I can run, for example, the code below without any errors:
```python
import duckdb
c = duckdb.connect()
print(c.execute("SELECT 42").fetchall())
```
It prints  `[(42,)]`

But if I try to run the code below I get the `NoSuchModuleError: Can't load plugin: sqlalchemy.dialects:duckdb` error:
```python
import ibis
con = ibis.duckdb.connect() 
con.list_tables()
```
What am I missing here?

**Comment by lostmygithubaccount** [comment]
Hi @rpmeira, which version of Ibis are you running? And can you confirm how you installed? 

`pip list | grep ibis`

The latest release is 5.1.0 and I'm not seeing this issue. You may need to `pip install 'ibis-framework[duckdb]'` per the conversation above. See https://ibis-project.org/backends/duckdb/#install for other options.

(As a side note, looks like the redirect from https://ibis-project.org/docs/3.2.0/backends/DuckDB/#install is broken, I think as a result of my recent PR.)

**Comment by rpmeira** [comment]
Hi @lostmygithubaccount. 
I was running version 5.1.0, but I had it installed with `pip install 'ibis-framework[sqlite]'`, as I was testing it with sqlite before. 
I didn't know I had do pip install it with every backend I wanted to used. I tought I just needed to install the actual backend package after the ibis was installed.

Now that I've installed with `pip install 'ibis-framework[sqlite]'` its wotking.
Thank you for the quick reply!
